### PR TITLE
[new release] bechamel-js, bechamel-notty, mperf, bechamel and bechamel-perf (0.1.0)

### PR DESCRIPTION
**Pull request by dinosaure**
HTML generator for bechamel's output

- Project page: <a href="https://github.com/dinosaure/bechamel">https://github.com/dinosaure/bechamel</a>
- Documentation: <a href="https://dinosaure.github.io/bechamel/">https://dinosaure.github.io/bechamel/</a>

##### CHANGES:

- First release

### CODE REVIEW

File: packages/bechamel-perf/bechamel-perf.0.1.0/opam [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+build: [
+  [ "dune" "subst" ] {pinned}
+  [ "dune" "build" "-p" name "-j" jobs ]
+  [ "dune" "runtest" "-p" name "-j" jobs ] {with-test}
+]
+
+depends: [
+  "ocaml"      {>= "4.08.0"}
+  "dune"       {>= "2.0.0"}
+  "mperf"
```
Comment (kit-ty-kate) [inline packages/bechamel-perf/bechamel-perf.0.1.0/opam:22]: Shouldn't this be also bound to the same version as the package?

**Comment by kit-ty-kate** [comment]
There are failures in `mperf` on OCaml < 4.07:
```
#=== ERROR while compiling mperf.0.1.0 ========================================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.02.3 | pinned(http://github.com/dinosaure/bechamel/releases/download/v0.1.0/bechamel-v0.1.0.tbz)
# path                 ~/.opam/4.02/.opam-switch/build/mperf.0.1.0
# command              ~/.opam/4.02/bin/dune build -p mperf -j 72
# exit-code            1
# env-file             ~/.opam/log/mperf-23-39e0b1.env
# output-file          ~/.opam/log/mperf-23-39e0b1.out
### output ###
#       ocamlc mperf/lib/.mperf.objs/byte/mperf.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.02/bin/ocamlc.opt -w -40 -g -bin-annot -I mperf/lib/.mperf.objs/byte -intf-suffix .ml -no-alias-deps -o mperf/lib/.mperf.objs/byte/mperf.cmo -c -impl mperf/lib/mperf.ml)
# File "mperf/lib/mperf.ml", line 122, characters 39-64:
# Error: Unbound value String.uncapitalize_ascii
#     ocamlopt mperf/lib/.mperf.objs/native/mperf.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.02/bin/ocamlopt.opt -w -40 -g -I mperf/lib/.mperf.objs/byte -I mperf/lib/.mperf.objs/native -intf-suffix .ml -no-alias-deps -o mperf/lib/.mperf.objs/native/mperf.cmx -c -impl mperf/lib/mperf.ml)
# File "mperf/lib/mperf.ml", line 122, characters 39-64:
# Error: Unbound value String.uncapitalize_ascii
```
```
#=== ERROR while compiling mperf.0.1.0 ========================================#
# context              2.0.7 | linux/x86_64 | ocaml-base-compiler.4.06.1 | pinned(http://github.com/dinosaure/bechamel/releases/download/v0.1.0/bechamel-v0.1.0.tbz)
# path                 ~/.opam/4.06/.opam-switch/build/mperf.0.1.0
# command              ~/.opam/4.06/bin/dune build -p mperf -j 72
# exit-code            1
# env-file             ~/.opam/log/mperf-23-39e0b1.env
# output-file          ~/.opam/log/mperf-23-39e0b1.out
### output ###
#       ocamlc mperf/lib/.mperf.objs/byte/mperf.{cmo,cmt} (exit 2)
# (cd _build/default && /home/opam/.opam/4.06/bin/ocamlc.opt -w -40 -g -bin-annot -I mperf/lib/.mperf.objs/byte -intf-suffix .ml -no-alias-deps -o mperf/lib/.mperf.objs/byte/mperf.cmo -c -impl mperf/lib/mperf.ml)
# File "mperf/lib/mperf.ml", line 204, characters 57-74:
# Error: Unknown builtin primitive "%caml_bytes_get64"
#     ocamlopt mperf/lib/.mperf.objs/native/mperf.{cmx,o} (exit 2)
# (cd _build/default && /home/opam/.opam/4.06/bin/ocamlopt.opt -w -40 -g -I mperf/lib/.mperf.objs/byte -I mperf/lib/.mperf.objs/native -intf-suffix .ml -no-alias-deps -o mperf/lib/.mperf.objs/native/mperf.cmx -c -impl mperf/lib/mperf.ml)
# File "mperf/lib/mperf.ml", line 204, characters 57-74:
# Error: Unknown builtin primitive "%caml_bytes_get64"
```
Looking at the other packages I'm guessing it is fine to change this directly here. I'll also update my patch upstream in a second.

**Comment by kit-ty-kate** [comment]
There was also an issue on Alpine:
```
#=== ERROR while compiling mperf.0.1.0 ========================================#
# context              2.0.7 | linux/x86_64 | ocaml-variants.4.11.0+spacetime | pinned(http://github.com/dinosaure/bechamel/releases/download/v0.1.0/bechamel-v0.1.0.tbz)
# path                 ~/.opam/4.11/.opam-switch/build/mperf.0.1.0
# command              ~/.opam/4.11/bin/dune build -p mperf -j 72
# exit-code            1
# env-file             ~/.opam/log/mperf-19-4aef26.env
# output-file          ~/.opam/log/mperf-19-4aef26.out
### output ###
#          gcc mperf/lib/mperf_stubs.o (exit 1)
# (cd _build/default/mperf/lib && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -fPIC -D_FILE_OFFSET_BITS=64 -D_REENTRANT -O2 -fno-strict-aliasing -fwrapv -fPIC -g -I /home/opam/.opam/4.11/lib/ocaml -o mperf_stubs.o -c mperf_stubs.c)
# mperf_stubs.c:4:10: fatal error: asm/unistd.h: No such file or directory
#  #include <asm/unistd.h>
#           ^~~~~~~~~~~~~~
# compilation terminated.

-          gcc mperf/lib/mperf_stubs.o (exit 1)
- (cd _build/default/mperf/lib && /usr/bin/gcc -O2 -fno-strict-aliasing -fwrapv -fPIC -D_FILE_OFFSET_BITS=64 -D_REENTRANT -O2 -fno-strict-aliasing -fwrapv -fPIC -g -I /home/opam/.opam/4.11/lib/ocaml -o mperf_stubs.o -c mperf_stubs.c)
- mperf_stubs.c:4:10: fatal error: asm/unistd.h: No such file or directory
-  #include <asm/unistd.h>
-           ^~~~~~~~~~~~~~
- compilation terminated.
```
I'm guessing this is the lack of linux headers, the fix is to require `conf-linux-libc-dev` I'm guessing

**Comment by dinosaure** [comment]
Ah thanks I did not deeply look into `perf` bindings, sorry.

File: packages/bechamel-perf/bechamel-perf.0.1.0/opam (hunk 1 above)
Comment (dinosaure) [reply packages/bechamel-perf/bechamel-perf.0.1.0/opam]: Hmmhmm, `mperf` will be not updated for a long time - so it should not follow the same release cycle of `bechamel`.

Comment (kit-ty-kate) [reply packages/bechamel-perf/bechamel-perf.0.1.0/opam]: ah my bad, I'll revert my change.

**Comment by kit-ty-kate** [comment]
Combining the last CI results and the one before it (the last commit should not have any effects), this seems all good. Thanks a lot!

**Pull request merged**